From d48449f1db77d96215f6b047e73548429d6b3d93 Mon Sep 17 00:00:00 2001
From: bbhtt <bbhtt.zn0i8@slmail.me>
Date: Sat, 29 Jul 2023 21:09:32 +0530
Subject: [PATCH] Add `$HOME/Downloads` as default download directory to aria2c
 and move it up the list

This enables tighter sandboxing as the default download directory is
`$HOME` and we do not want to give access to entire home directory by
default
---
 src/ui/preferences_dialog.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/ui/preferences_dialog.c b/src/ui/preferences_dialog.c
index 11d471ba..35815068 100644
--- a/src/ui/preferences_dialog.c
+++ b/src/ui/preferences_dialog.c
@@ -66,16 +66,16 @@ static PreferencesDialog *prefdialog = NULL;

 /** download tool commands need to take an URI as %s */
 static const gchar * enclosure_download_commands[] = {
+	"/bin/sh -c 'aria2c -d ${XDG_DOWNLOAD_DIR:-$HOME/Downloads} %s'",
 	"steadyflow add %s",
 	"dbus-send --session --print-reply --dest=org.gnome.gwget.ApplicationService /org/gnome/gwget/Gwget org.gnome.gwget.Application.OpenURI string:%s uint32:0",
 	"kget %s",
 	"uget-gtk %s",
-	"transmission-gtk %s",
-	"aria2c %s"
+	"transmission-gtk %s"
 };

 /** order must match enclosure_download_commands[] */
-static const gchar *enclosure_download_tool_options[] = { "Steadyflow", "gwget", "KGet", "uGet", "Transmission (Gtk)", "aria2", NULL };
+static const gchar *enclosure_download_tool_options[] = { "aria2", "Steadyflow", "gwget", "KGet", "uGet", "Transmission (Gtk)", NULL };

 /** GConf representation of toolbar styles */
 static const gchar * gui_toolbar_style_values[] = { "", "both", "both-horiz", "icons", "text", NULL };
--
2.41.0

